### ISSUE TITLE
Fast, Inaccurate Mode

### ISSUE THREAD

**Issue by WolfgangWaltenberger** [scikit-hep/pyhf#1056]
# Description

We SModelS kindly wish to propose some kind of a fast mode for pyhf, sacrificing O(10%) accuracy on the likelihoods for increase in speed. We SModelS would be interested in such a feature. And we would hope that use cases other than ours would also benefit from such a mode. We are thinking of use cases where accuracy is not the top priority. Anyways, just a proposal, just think about it! Cheers, Wolfgang, for SModelS

**Comment by lukasheinrich** [comment]
Hi @WolfgangWaltenberger,

one thing that should be possible once we have full likelihoods is to then follow the same procedure as CMS by providing "background-integrated" covariance matrices for the various regions. We need some of the error propagation machinery needed also by @alexander-held and @ntadej which needs to go into pyhf but at that point this can be very fast at the cost precisino.

**Comment by WolfgangWaltenberger** [comment]
On Tue, 8 Sep 2020 at 18:51, Lukas <notifications@github.com> wrote:

> Hi @WolfgangWaltenberger <https://github.com/WolfgangWaltenberger>,
>
> one thing that should be possible once we have full likelihoods is to then
> follow the same procedure as CMS by providing "background-integrated"
> covariance matrices for the various regions. We need some of the error
> propagation machinery needed also by @alexander-held
> <https://github.com/alexander-held> and @ntadej
> <https://github.com/ntadej> which needs to go into pyhf but at that point
> this can be very fast at the cost precisino.
>
By background-integrated covariance matrices, you have in mind one big
Gaussian that "envelops" all nuisances, right? Yes, I was also thinking in
that direction.
In this case it would esssentially be what in CMS is called a simplified
likelihood, discussed e.g. here:

https://inspirehep.net/literature/1694152

It might result in nice spin-offs, like a good initialization algorithm for
e.g. maximizing the full likelihood, etc.

Wolfgang

**Comment by WolfgangWaltenberger** [comment]
Ah Sabine reminded me of one more thing. If you go for this Gaussian-envelope approach, then we strongly recommend that you add a skewness term, or something similar to account for asymmetries. From our experience a Gaussian alone is too crude a model.

**Comment by lukasheinrich** [comment]
yes it was always the plan that once we have the full likelihoods that the community could/would be able to develop lossy versions of it. To some extent `pyhf prune` which allows removing systematic terms etc  makes this already possible

**Comment by sabinekraml** [comment]
Hi, may I ask what "prune" is really doing? Does it literally remove uncertainty contributions (which is not whet we want) or integrate over them?

**Comment by matthewfeickert** [comment]
> Hi, may I ask what "prune" is really doing? Does it literally remove uncertainty contributions (which is not whet we want) or integrate over them?

The docs on [`pyhf.workspace.Workspace.prune`](https://scikit-hep.org/pyhf/_generated/pyhf.workspace.Workspace.html#pyhf.workspace.Workspace.prune) probably make this the most clear, but it is just trimming things that you specify off of the workspace.

**Comment by sabinekraml** [comment]
OK, thanks. So this is indeed not what we want for a fast mode. Instead of having things trimmed off, we'd want them to be profiled or marginalized over. That goes in the direction of the "Gaussian envelope" Wolfgang was mentioning above, but without loosing relevant asymmetries (i.e. keeping the skew).

**Comment by sabinekraml** [comment]
well, of course nuisances, which are really "irrelevant" from a fast mode point of view, may also be removed. I just want to say that it will be nice to go a bit further.

**Comment by matthewfeickert** [comment]
This feedback and discussion is great so far &mdash so thanks. I think given the timescale that we want to get `v0.5.3` out on we probably won't be able to get this in. We should try to follow up more on this for `v0.6.0` which should be the next release after `v0.5.3` and we can see if it is reasonable to get in there.
